### PR TITLE
Them tiers: Refactor theme tier selectors

### PR DESCRIPTION
**Pull request by rcrdortiz**
Related to:
- https://github.com/Automattic/wp-calypso/issues/85899

## Proposed Changes

* Created separate selectors for `isThemeAllowedOnSite` and `getThemeTierForTheme`.
* Deleted the old hook and selector.
* Updated references to use the new selectors.

## Testing Instructions

* Open the Calypso Live link
* Goto `/themes`
* Add the `?flags=themes/tiers' flag
* Check that all the badges and everything else works as expected.

No before and after screenshots provided since there are no visual changes.
## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by rcrdortiz** [comment]
This PR is now ready for review and no longer WiP.

File: client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js [hunk 1]
```diff
@@ -24,14 +24,14 @@ export default function ThemeTierPartnerBadge() {
 	const subscriptionPrices = useSelector( ( state ) =>
 		getMarketplaceThemeSubscriptionPrices( state, themeId )
 	);
-	const { isThemeAllowedOnSite } = useThemeTier( siteId, themeId );
+	const isThemeAllowed = useSelector( ( state ) => isThemeAllowedOnSite( state, themeId ) );
 
-	const labelText = isThemeAllowedOnSite
+	const labelText = isThemeAllowed
```
Comment (BogdanUngureanu) [inline client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js:29]: Nitpick: if we keep the `isThemeAllowed` const, we don't need to rename it in multiple places.

File: client/components/theme-tier/theme-tier-badge/index.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
 import { BUNDLED_THEME, DOT_ORG_THEME, MARKETPLACE_THEME } from '@automattic/design-picker';
 import { useSelector } from 'calypso/state';
-import { getThemeType, isThemePurchased } from 'calypso/state/themes/selectors';
+import {
+	getThemeType,
+	isThemePurchased,
+	getThemeTierForTheme,
+	isThemeAllowedOnSite,
```
Comment (BogdanUngureanu) [inline client/components/theme-tier/theme-tier-badge/index.js:7]: we can name import with an alias? 🤔

File: client/state/themes/test/selectors.js [hunk 3]
```diff
@@ -66,6 +68,13 @@ const twentyfifteen = {
 	author_uri: 'https://wordpress.org/',
 };
 
+const freeThemeTier = {
+	slug: 'free',
+	name: 'Free',
+	plan: PLAN_FREE,
+	feature: FEATURE_WOOP,
```
Comment (BogdanUngureanu) [inline client/state/themes/test/selectors.js:75]: Why do we use `FEATURE_WOOP` and not its real feature (null)? 🤔

Comment (rcrdortiz) [reply client/state/themes/test/selectors.js:75]: It's a test, I would use ANY_FEATURE but we don't have one XD. When testing we don't care about a given specific feature, and it should work with any feature.

File: client/components/theme-tier/theme-tier-badge/index.js (hunk 2 above)
Comment (rcrdortiz) [reply client/components/theme-tier/theme-tier-badge/index.js:7]: Yeah I agree, but I would need to change everything again and we wouldn't get much benefit.

**Review by BogdanUngureanu — APPROVED** [review]
Works as expected now! :shipit:

**Pull request merged**